src/compressor/Compressor.cc

```diff
@@ -45,6 +48,10 @@ boost::optional<Compressor::CompressionAlgorithm> Compressor::get_comp_alg_type(
 #ifdef HAVE_LZ4
   if (s == "lz4")
     return COMP_ALG_LZ4;
+#endif
+#ifdef HAVE_QATZIP
+  if (s == "qatzip")
+    return COMP_ALG_QATZIP;
```

Comment:
So my main question here is the same as on the other pull request: is the QAT compressed data byte-compatible with the unaccellerated versions?  If so, then I don't think this should be a different compression type/plugin, but should instead transparently accelerate the existing compression types.  That way the on-disk structures all still say lz4 or zlib but they just go faster. And a simple code change will make existing compressed data decompress faster (vs only having newly compressed data go faster).

Reply:
Currently QAT can accelerate several algorithms, including deflate, lz4 and snappy, and it should be data byte-compatible with unaccellerated versions in current several plugins. But qatzip has different API with libsnappy/zlib/lz4 library, could not almost share any code with current plugins. In other words, it has different work flow and it also have to initialize and setup QAT hardware. If we want to transparently accelerate the existing compression types, we should have to add one common tool class for qatzip library, and call this class to initialize QAT hardware when those plugins supported by QAT construct, and then no any next code can be shared. So, compared with this, I guess that one individual plugin for QAT maybe clearer and won't change current code framework. What do you think about this?

Reply:
In addition, QAT can support more algorithms in future, if one individual plugin, the code won't change anything. If no, other plugins will have to make more changes at that time.

Reply:
@liewegas what do you think about my reply?

Reply:
@tchaikov sounds good, I will try this way.

Reply:
The key thing is that code in e.g. bluestore does htis:
```
  int alg = int(chdr.type);
  CompressorRef cp = compressor;
  if (!cp || (int)cp->get_type() != alg) {
    cp = Compressor::create(cct, alg);
  }
```
That is, we need to return a Compressor object based on an on-disk field that is one of
```
    COMP_ALG_NONE = 0,
    COMP_ALG_SNAPPY = 1,
    COMP_ALG_ZLIB = 2,
    COMP_ALG_ZSTD = 3,
#ifdef HAVE_LZ4
    COMP_ALG_LZ4 = 4,
#endif
#ifdef HAVE_BROTLI
    COMP_ALG_BROTLI = 5,
#endif
```
So if the on-disk structure is e.g. ZLIB, we should try to return something that will use QAT.. but if not fall back to the normal zlib code.

Whether this is structured as an attempt to load a QAT class in different modes or combined with the existing zlib and lz4 classes is a detail I'm not too concerned about.  Though my guess a toolkit class to do the is the latter is a bit cleaner...